src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -128,15 +89,56 @@ uct_cuda_ipc_mem_add_reg(void *addr, uct_cuda_ipc_memh_t *memh,
 
         key->ph.handle_type = UCT_CUDA_IPC_KEY_HANDLE_TYPE_VMM;
         ucs_trace("packed vmm fabric handle for %p", addr);
+        goto common_path;
+    } else {
+        status = UCT_CUDADRV_FUNC_LOG_ERR(cuPointerGetAttribute(&mempool,
+                    CU_POINTER_ATTRIBUTE_MEMPOOL_HANDLE, (CUdeviceptr)addr));
+        if (status == UCS_OK) {
+            status = UCT_CUDADRV_FUNC_LOG_ERR(cuPointerGetAttribute(
+                        &allowed_handle_types,
+                        CU_POINTER_ATTRIBUTE_ALLOWED_HANDLE_TYPES,
+                        (CUdeviceptr)addr));
+            if ((status != UCS_OK) ||
+                (!(allowed_handle_types | CU_MEM_HANDLE_TYPE_FABRIC))) {
```

Comment:
should it be & here instead of | ?
if yes, seems this code was not tested, can we add test with different cuda mempool types?